tap_google_ads/client.py

```diff
@@ -1,17 +1,114 @@
+    """
+    if service_account_info:
+        return service_account.Credentials.from_service_account_info(json.loads(service_account_info), subject=impersonated_email, scopes=['https://www.googleapis.com/auth/adwords'])
+    else:
+        raise ValueError("GOOGLE_ADS_SERVICE_ACCOUNT_INFO_STRING environment variable not set")
+
+class GoogleAdsClientServiceAccount(GoogleAdsClient):
+    @classmethod
+    def _get_client_kwargs(cls, config_data):
+        """Converts configuration dict into kwargs required by the client.
+
+        Args:
+            config_data: a dict containing client configuration.
+
+        Returns:
+            A dict containing kwargs that will be provided to the
+            GoogleAdsClientServiceAccount initializer.
+
+        Raises:
+            ValueError: If the configuration lacks a required field.
+        """
+        return {
+            "credentials": get_service_account_credentials(config_data.get("json_key_file_path"), config_data.get("impersonated_email")),
+            "developer_token": config_data.get("developer_token"),
+            "endpoint": config_data.get("endpoint"),
+            "login_customer_id": config_data.get("login_customer_id"),
+            "logging_config": config_data.get("logging"),
+            "linked_customer_id": config_data.get("linked_customer_id"),
+            "http_proxy": config_data.get("http_proxy"),
+            "use_proto_plus": config_data.get("use_proto_plus"),
+            "use_cloud_org_for_api_access": config_data.get(
+                "use_cloud_org_for_api_access"
+            ),
+        }
+
+
+def create_sdk_client(config, login_customer_id=None, auth_method=None):
+    """
+    Determines configuration dictionary based on authentication method.
+
+    Args:
+        config_data: a dict containing client configuration.
+
+    Returns:
+        GoogleAdsClientServiceAccount.
+    """
+    service_account_info = os.getenv("GOOGLE_ADS_SERVICE_ACCOUNT_INFO_STRING")
+    impersonated_email = os.getenv("GOOGLE_ADS_IMPERSONATED_EMAIL")
+    
+    if config.get("auth_method") == "Service_Account":
+        CONFIG = {
+            "use_proto_plus": config["use_proto_plus"],
+            "developer_token": config["developer_token"],
+            "impersonated_email": impersonated_email,
+            "json_key_file_path": service_account_info,
+        }
+    else:
+        CONFIG = {
+            "use_proto_plus": config["use_proto_plus"],
+            "developer_token": config["developer_token"],
+            "client_id": config["oauth_client_id"],
+            "client_secret": config["oauth_client_secret"],
+            "refresh_token": config["refresh_token"],
+            "login_customer_id": config["login_customer_id"],
+        }
 
     if login_customer_id:
         CONFIG["login_customer_id"] = login_customer_id
 
-    sdk_client = GoogleAdsClient.load_from_dict(CONFIG)
+    sdk_client = GoogleAdsClientServiceAccount.load_from_dict(CONFIG)
```

Comment:
I would suggest using the name `GoogleAdsClientServiceAccountInfo` for the class, since we are adding the functionality to support the function `service_account.Credentials.from_service_account_info()`